src/main/drivers/flash_w25n.c

```diff
@@ -313,30 +324,34 @@ static bool w25n01g_waitForReady(flashDevice_t *fdevice)
  * The flash requires this write enable command to be sent before commands that would cause
  * a write like program and erase.
  */
-static void w25n01g_writeEnable(flashDevice_t *fdevice)
+static void w25n_writeEnable(flashDevice_t *fdevice)
 {
-    w25n01g_performOneByteCommand(&fdevice->io, W25N01G_INSTRUCTION_WRITE_ENABLE);
+    w25n_performOneByteCommand(&fdevice->io, W25N_INSTRUCTION_WRITE_ENABLE);
 
     // Assume that we're about to do some writing, so the device is just about to become busy
     fdevice->couldBeBusy = true;
 }
 
-const flashVTable_t w25n01g_vTable;
+const flashVTable_t w25n_vTable;
 
-bool w25n01g_identify(flashDevice_t *fdevice, uint32_t jedecID)
+bool w25n_identify(flashDevice_t *fdevice, uint32_t jedecID)
 {
-    switch (jedecID) {
-    case JEDEC_ID_WINBOND_W25N01GV:
-        fdevice->geometry.sectors = 1024;      // Blocks
-        fdevice->geometry.pagesPerSector = 64; // Pages/Blocks
-        fdevice->geometry.pageSize = 2048;
-        break;
+    flashGeometry_t *geometry = &fdevice->geometry;
+    uint8_t index;
+
+    for (index = 0; w25nFlashConfig[index].jedecID; index++) {
```

Comment:
Or 
```suggestion
    for (index = 0; index < ARRAYLEN(w25nFlashConfig); index++) {
```
but both will do.

You can also put it into `static` find function, returning pointer (that will make 'not found' branch cleaner)